### ISSUE TITLE
Weird rotation in iPad?

### ISSUE THREAD

**Issue by eflyjason** [ealeksandrov/EAIntroView#149]
So here is my code for creating the `EAIntroView` (using Swift)

```
    let titleColor = UIColor.blackColor()
    let descColor = UIColor.blueColor()

    let page1 = EAIntroPage()
    page1.title = "Some title"
    page1.desc = "Some deep"
    page1.titleColor = titleColor
    page1.descColor = descColor
    page1.bgImage = UIImage(named: "bg1")

    let page2 = EAIntroPage()
    page2.title = "page 2"
    page2.desc = "something again"
    page2.titleColor = titleColor
    page2.descColor = descColor

    let page3 = EAIntroPage()
    page3.title = "page 3"
    page3.desc = "again again again"
    page3.titleColor = titleColor
    page3.descColor = descColor

    let intro = EAIntroView(frame: self.view.bounds, andPages: [page1,page2,page3])
    intro.delegate = self
    intro.pageControl.pageIndicatorTintColor = UIColor.blackColor()
    intro.pageControl.currentPageIndicatorTintColor = UIColor.redColor()

    intro.showInView(self.view, animateDuration: 0.5)
```

But on my iPad, the following display in iPad:

![untitled](https://cloud.githubusercontent.com/assets/4939421/11767167/40b4a746-a1df-11e5-9162-0382418f2e9c.gif)

How can I fix it? Thanks!

**Comment by eflyjason** [comment]
Also, it works fine on iPhone 6 Plus

**Comment by ealeksandrov** [comment]
Thanks for great demo of bug! I will take a look into it on a week.

**Comment by fozoglu** [comment]
Hi @ealeksandrov, I want to use in my iPad project. When this problem is solved? Thanks.

**Comment by vladdaneliuc** [comment]
i have the same issue

**Comment by fabioprodoctor** [comment]
Same issue here. 

Does anyone know how to block rotate? I think this can be one workaround for this issue

**Comment by ealeksandrov** [comment]
Can anyone confirm this bug on **device**?
I struggled with debugging on simulator until tested it on my iPad 4. And it worked flawlessly, just as designed.

**Comment by ealeksandrov** [comment]
Tested:
iOS 9.3 simulators:
* iPad 2 - working
* iPad Retina - working
* iPad Air - broken
* iPad Air 2 - broken
* iPad Pro - broken

Devices:
* iPad 4 - working

Since I can't reproduce this on on my device and there is no possible reason for this behaviour - I consider it as iOS Simulator bug.

**Comment by fozoglu** [comment]
@ealeksandrov
My Device:
iPad Air (First Generation) - working

Thanks.

**Comment by ealeksandrov** [comment]
Thanks for reporting!
Closing as iOS Simulator bug.

**Comment by eflyjason** [comment]
iPad Air (First Generation) - broken :(

**Comment by ealeksandrov** [comment]
O_o
File a radar. I really can't see any reason for this behaviour. This is iOS autolayout issue.

**Comment by gunnarblom** [comment]
I have the same problem. It seems the layout doesn't change so e.g. the portrait layout is still there even if you switch to landscape. Verified on iPad Air 2.

@ealeksandrov Can you block the rotation?

**Comment by ealeksandrov** [comment]
Will look for devices to test it.
Try to override `shouldAutorotate` in VC containing EAIntroView.

**Comment by SnyersK** [comment]
I can confirm this occurs with version 2.9.0 on an iPad Air running iOS 9.3.2.
It does change the layout on rotation, but it seems it shows the portrait orientation when rotating to landscape and vice versa.

A 'sort of' workaround is adding the pages again to EAIntroView after the rotation, but the animation remains incorrect of course.

It does however show the correct layout when first opening the intro view on any rotation.

**Comment by SnyersK** [comment]
I can't seem to reproduce this bug using your example app on the same iPad Air.
Tried out a couple of things, it doesn't happen.

I fail to see what's different in my project though.

I'm using EAIntroView in a presented ViewController.
This works perfectly in the example application.

**Comment by ealeksandrov** [comment]
Weird. Thanks for info, lets collect more feedback in this issue.

**Comment by ealeksandrov** [comment]
Can anyone update us on this bug with iOS 10? Particularly with projects and devices already experienced this issue on iOS 9.

**Comment by bahrio** [comment]
Experiencing this problem on the iOS 10 simulator (for iPad Air)

**Comment by kee23** [comment]
Experiencing it on actual device as well (iPad Air 2) using iOS 9.3.1. Same as above, in that example works fine, but in project it does not.

I'll update to iOS 10 later and let you know if anything changes.

**Comment by algrid** [comment]
I also can reproduce this in iOS 10.1 simulator.

It looks like the problem is in ```UIDeviceOrientationDidChangeNotification``` handler. You're expecting there that all the frames have already changed after rotation. But it isn't always the case. An alternative would be to override ```layoutSubviews``` and do all the changes there.

**Comment by ealeksandrov** [comment]
Fixed by #191
